src/query/service/src/pipelines/processors/transforms/hash_join/join_hash_table.rs

```diff
@@ -542,10 +543,21 @@ impl HashJoinState for JoinHashTable {
         }
 
         let mut chunks = self.row_space.chunks.write().unwrap();
+        let mut has_null = false;
         for chunk_index in 0..chunks.len() {
             let chunk = &mut chunks[chunk_index];
             let mut columns = Vec::with_capacity(chunk.cols.len());
             let markers = if self.hash_join_desc.join_type == Mark {
+                if self.hash_join_desc.marker_join_desc.subquery_as_build_side && !has_null {
+                    if let Some(validity) = chunk.cols[0].validity().1 {
```

Comment:
Here, `cols` in `chunk` are `build_keys`(join attributes for build side), if any column in `cols` has null, `has_null` will be set to true.